File: LivrusVirtus/KDevelop/wxWin/ODBC/GTK/plivrusgtk/Splash.cpp
#include "resource.h"
#include "Splash.h"
#include "Login.h"
#include "Principal.h"

#include "xpms/livrusvirtus.xpm"

BEGIN_EVENT_TABLE(CfrmSplash, wxSplashScreen)
    EVT_CLOSE(CfrmSplash::OnClose)
END_EVENT_TABLE()

CfrmSplash::CfrmSplash(wxWindow *parent) 
    : wxSplashScreen(//wxBitmap(BMP_LIVRUSVIRTUS, wxBITMAP_TYPE_BMP),
          wxBitmap(livrusvirtus_xpm),
          wxSPLASH_CENTRE_ON_SCREEN|wxSPLASH_TIMEOUT,
          3000, parent, -1, wxDefaultPosition, wxDefaultSize,
          wxSTAY_ON_TOP)
{ }

CfrmSplash::~CfrmSplash() { }

void CfrmSplash::OnClose(wxCloseEvent& event){      
	frmLogin = new CfrmLogin((wxWindow *)NULL);
	frmLogin->Show(true);
	event.Skip();   
}

File: LivrusVirtus/BoaConstructor/wxPython/Splash.py
#Boa:SplashScreen:frmSplash

from wxPython.wx import *

import Login

def create(parent):
    return frmSplash(parent)

[wxID_FRMSPLASH] = map(lambda _init_ctrls: wxNewId(), range(1))

class frmSplash(wxSplashScreen):
    def _init_utils(self):
        pass
    
    def _init_ctrls(self, prnt):
        wxSplashScreen.__init__(self, bitmap=wxBitmap(u'LivrusVirtus.bmp',
          wxBITMAP_TYPE_BMP), splashStyle=wxSPLASH_CENTRE_ON_SCREEN | wxSPLASH_TIMEOUT,
          milliseconds=1000, parent=prnt, id=wxID_FRMSPLASH, 
          pos=wxDefaultPosition, size=wxDefaultSize,
          style=wxSIMPLE_BORDER | wxSTAY_ON_TOP)
        EVT_CLOSE(self, self.OnFrmsplashClose)
        
    def __init__(self, parent):
        self._init_ctrls(parent)

    def OnFrmsplashClose(self, event):        
        fLogin = Login.create(None)
        fLogin.Show()
        event.Skip()
